pennylane/qaoa/qaoa.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This file contains built-in functions for defining QAOA workflows
+"""
+from collections.abc import Iterable
+import networkx
+import pennylane as qml
+from pennylane.wires import Wires
+
+
+def _check_iterable_graph(graph):
+    """ Checks if a graph supplied in 'Iterable format' is formatted correctly
+
+        Args:
+            graph (Iterable): The graph that is being checked
+    """
+
+    for g in graph:
+
+        if not isinstance(g, Iterable):
+            raise ValueError(
+                "Elements of `graph` must be Iterable objects, got {}".format(type(g).__name__)
+            )
+        if len(g) != 2:
+            raise ValueError(
+                "Elements of `graph` must be Iterable objects of length 2, got length {}".format(
+                    len(g)
+                )
+            )
+        if g[0] == g[1]:
+            raise ValueError("Edges must end in distinct nodes, got {}".format(g))
+
+    if len({tuple(g) for g in graph}) != len(graph):
+        raise ValueError("Nodes cannot be connected by more than one edge")
+
+
+##################### Mixer Hamiltonians #####################
+
+
+def x_mixer(wires):
+    r"""Creates the basic Pauli-X mixer Hamiltonian used in the original `QAOA paper <https://arxiv.org/abs/1411.4028>`__.
+    This Hamiltonian is defined as:
+
+    .. math:: H_M \ = \ \displaystyle\sum_{i} X_{i},
+
+    where :math:`i` ranges over all qubits, and :math:`X_i` denotes the Pauli-X operator on the :math:`i`-th qubit (wire).
```

Comment:
I recommend we just use 'wire(s)' everywhere, rather than swapping between qubits/wires, for consistency with the rest of the documentation.

Reply:
(this might apply elsewhere in case I missed it)